src/team_lib/mhba/xccl_mhba_lib.h

```diff
@@ -11,6 +11,9 @@
 #include <infiniband/verbs.h>
 #include <infiniband/mlx5dv.h>
 
+#define MAX_CONCURRENT_OUTSTANDING_ALL2ALL 52 //todo change - according to limitations
```

Comment:
why 52? lets not move close to the HW limit. In reality, it is very rare even to have just 2 outstanding concurrent non-blocking collectives. I would say, for our purposes, 4 is enough (mostly for testing)